### DISCUSSION TITLE
createSchemaField 方法是否允许传第三方扩展参数

### DISCUSSION THREAD

**Discussion by tgxpuisb** [alibaba/formily#1578] [Ideas]
业务场景：第三方定义的图片上传组件中的action参数需要根据项目处于不同的环境 dev 、online 做出动态的调整，所以希望是可以在引用组件的地方将数据透传到第三方组件中

![image](https://user-images.githubusercontent.com/11400784/122004444-00d48800-cde7-11eb-8030-df4078ca2f9c.png)

通过 useContext 是可以获取的，实现上是没有什么大问题的

![image](https://user-images.githubusercontent.com/11400784/122004504-12b62b00-cde7-11eb-9896-7b944200aca8.png)

但是 从TS的角度来看貌似 createSchemaField 只能传入 component 和 scope 两个key对应的参数，所以想问一下这是否是官方推荐的做法

**Comment by janryWang** [comment]
推荐自己用react context往下透传，你这种用法有点trick了

**Reply by tgxpuisb** [reply]
好的，收到
